src/main/java/me/contaria/seedqueue/mixin/server/MinecraftServerMixin.java

```diff
@@ -183,9 +183,18 @@ private void pauseServer(MinecraftServer server, boolean value, Operation<Void>
         if (!entry.hasWorldPreview()) {
             return false;
         }
-        if (!entry.isLocked() && SeedQueue.config.maxWorldGenerationPercentage < 100) {
+        if (entry.isLocked()) {
+            return false;
+        }
+        if (SeedQueue.config.resumeOnFilledQueue && SeedQueue.isFull() && entry.isMaxWorldGenerationReached()) {

```

Comment:
```suggestion
        if (SeedQueue.config.resumeOnFilledQueue && entry.isMaxWorldGenerationReached() && SeedQueue.isFull()) {
```
Please swap the order of these statements, I'm a little worried about performance issues from the `SeedQueue#isFull` check but if it only gets executed once the entry has actually reached max world generation, been paused and then resumed I think it should be fine

Reply:
I'm sure it does happen only once.... Its like... It loads all of the queue completely first then checks each seed in queue in order to see which fits the condition one by one and unpauses them in order... Though idk if these checks are repeated again when the unpause happens... I'm assuming not but I'm a little sceptical

Reply:
`seedQueue$shouldPause` is run constantly on the server thread, which is why i want to avoid calling synchronized code/code that iterates over the queue in there as much as possible

Reply:
Hmm okay fair... I'll think of a solution to it

Reply:
Oh wait i made a mistake here, this isn't synchronized on the list since it's just getting the count. I'd still like the the conditions to be swapped but it's not gonna be a performance issue since it only has to get the count of the `LinkedBlockingQueue` from an `AtomicInteger`

Reply:
Oh yeah we aren't iterating through the Queue but just getting a property of it... Lol fair ok I'll make those changes and request review again tomorrow :D